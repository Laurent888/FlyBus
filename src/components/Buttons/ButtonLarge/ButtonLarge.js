import React from "react";
import "./ButtonLarge.scss";

const ButtonLarge = props => {
  return (
    <div className={`btn-large ${props.btnType}`}>
      <button onClick={props.clicked}>{props.label}</button>
    </div>
  );
};

export default ButtonLarge;
