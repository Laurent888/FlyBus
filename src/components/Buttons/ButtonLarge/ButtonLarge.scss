@import "../../../variables";

.btn-large {
  button {
    box-sizing: border-box;
    border-radius: 5px;
    padding: 1.3rem 2.5rem;
    font-size: 1.6rem;
    font-weight: 700;
    letter-spacing: 1px;
    text-transform: uppercase;
    outline: none;
    margin: 2rem;

    @media screen and (max-width: $breakpoint-width) {
      margin: 1rem;
    }
  }
  &.danger {
    button {
      color: #fff;
      border: 1px solid #e00d57;
      background-color: #e00d57;
      transition: all 0.2s ease-out;
      &:hover {
        background-color: #9a0036;
        border: 1px solid #9a0036;
        cursor: pointer;
      }
      &:active {
        background-color: #9a00369d;
      }
    }
  }

  &.default {
    button {
      color: $primary-color;
      border: 1px solid rgb(197, 197, 197);
      background-color: #ffffff;
      transition: all 0.2s ease-out;
      &:hover {
        background-color: #cecece;
        cursor: pointer;
      }
      &:active {
        background-color: #cecece94;
      }
    }
  }
  &.success {
    button {
      color: #fff;
      border: 1px solid #37b937;
      background-color: #37b937;
      transition: all 0.2s ease-out;
      &:hover {
        background-color: #30a030;
        cursor: pointer;
      }
      &:active {
        background-color: #30a0306c;
      }
    }
  }
  @media screen and (max-width: $breakpoint-width) {
    button {
      padding: 0.8rem 1.7rem;
      font-size: 1.3rem;
    }
  }
}
