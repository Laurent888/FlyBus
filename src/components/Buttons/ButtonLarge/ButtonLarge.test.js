import React from "react";
import { shallow } from "enzyme";
import ButtonLarge from "./ButtonLarge";

describe("Button Large", () => {
  const btnType = "btnPrimary";
  const label = "test";
  const clicked = () => console.log("ok");

  const wrapper = shallow(
    <ButtonLarge btnType={btnType} label={label} clicked={clicked} />
  );

  it("should render button large", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have TEXT in the label", () => {
    expect(wrapper.text()).toBe("test");
  });
});
