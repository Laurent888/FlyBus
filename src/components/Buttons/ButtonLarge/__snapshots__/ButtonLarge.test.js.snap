// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Large should render button large 1`] = `
<div
  className="btn-large btnPrimary"
>
  <button
    onClick={[Function]}
  >
    test
  </button>
</div>
`;
